### PR TITLE
warn for `iter.next()`

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Warn for next() -> __next__()

```
py2.x:

 def next(self):
        try:
            f = self.files.pop(0)
        except IndexError:
            raise StopIteration
        else:
            s = self.server.avatar._runAsUser(
                os.lstat, os.path.join(self.dir, f))
            longname = lsLine(f, s)
            attrs = self.server._getAttrs(s)
            return (f, longname, attrs)

py3.x:

 def __next__(self):
        try:
            f = self.files.pop(0)
        except IndexError:
            raise StopIteration
        else:
            s = self.server.avatar._runAsUser(
                os.lstat, os.path.join(self.dir, f))
            longname = lsLine(f, s)
            attrs = self.server._getAttrs(s)
            return (f, longname, attrs)

    next = __next__ 

python 2.x:

>>> marks = [65, 71, 68, 74, 61]
>>> iterator_marks = iter(marks)
>>> marks_1 = next(iterator_marks)
>>> iterator_marks.next()
71
>>> iterator_marks.__next__()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'listiterator' object has no attribute '__next__'
>>> 

python 3.x:

>>> marks = [65, 71, 68, 74, 61]
>>> iterator_marks = iter(marks)
>>> marks_1 = next(iterator_marks)
>>> marks_1 = __next__(iterator_marks)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name '__next__' is not defined
>>> iterator_marks.next()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list_iterator' object has no attribute 'next'
>>> iterator_marks.__next__()
71
```

### CODE REVIEW

File: Lib/test/test_py3kwarn.py [hunk 1]
```diff
@@ -206,6 +206,21 @@ def test_frame_attributes(self):
                 w.reset()
                 self.assertWarning(setattr(f, attr, None), w, expected)
 
+    def test_listiter_next(self):
+        expected = "the attribute 'next' is not supported in 3.x: use '__next__' or create a 'next' alias"
```
Comment (ltratt) [inline Lib/test/test_py3kwarn.py:210]: I think we mean "The global function `next`"?

Comment (nanjekyejoannah) [reply Lib/test/test_py3kwarn.py:210]: The global builtin works:

py2.x:

```
>>> marks = [65, 71, 68, 74, 61]
>>> iterator_marks = iter(marks)
>>> marks_1 = next(iterator_marks)
>>> iterator_marks.next()
71
```

py3.x:

```
>>> marks = [65, 71, 68, 74, 61]
>>> iterator_marks = iter(marks)
>>> marks_1 = next(iterator_marks)
>>> iterator_marks.next()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list_iterator' object has no attribute 'next'
```

its the `internext()` that fails:

`x = [2, 3]`

This works:

`next(iter(x))`

This wont:

`iter(x).next()`

unless its changed to:

`iter(x).__next__()`

Comment (ltratt) [reply Lib/test/test_py3kwarn.py:210]: Oh, I see!

**Comment by nanjekyejoannah** [comment]
@ltratt If this PR gets approved I will use the same warning implementation in https://github.com/softdevteam/pygrate3/pull/13

**Comment by nanjekyejoannah** [comment]
Closing this in favour of https://github.com/softdevteam/pygrate3/pull/13

**Pull request closed without merging**